tests/test_snapshot.py

```diff
@@ -0,0 +1,128 @@
+import unittest
+from unittest.mock import patch
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from threading import Thread
+
+import httpretty
+from selenium import webdriver
+from selenium.webdriver.firefox.options import Options as FirefoxOptions
+
+from percy import percySnapshot
+import percy.snapshot as local
+LABEL = local.LABEL
+
+# mock a simple webpage to snapshot
+class MockServerRequestHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(200)
+        self.send_header('Content-Type', 'text/html')
+        self.end_headers()
+        self.wfile.write(('Snapshot Me').encode('utf-8'))
+    def log_message(self, format, *args):
+        return
+
+# daemon threads automatically shut down when the main process exits
+mock_server = HTTPServer(('localhost', 8000), MockServerRequestHandler)
+mock_server_thread = Thread(target=mock_server.serve_forever)
+mock_server_thread.setDaemon(True)
+mock_server_thread.start()
+
+# mock helpers
+def mock_healthcheck(fail=False):
+    # rather than a healthcheck, this SDK tries to fetch @percy/dom
+    httpretty.register_uri(
+        httpretty.GET, 'http://localhost:5338/percy/dom.js',
+        # mock serialization script for testing purposes
+        body='window.PercyDOM = { serialize: () => document.documentElement.outerHTML };',
+        status=(500 if fail else 200))
+
+def mock_snapshot(fail=False):
+    httpretty.register_uri(
+        httpretty.POST, 'http://localhost:5338/percy/snapshot',
+        body=('{ "success": ' + ('true' if not fail else 'false, "error": "test"') + '}'),
+        status=(500 if fail else 200))
+
+class TestPercySnapshot(unittest.TestCase):
```

Comment:
Consider `pytest` for more readable tests :)

Reply:
In all honesty, I went with what was easiest to set up as a python newbie. Having these tests at all is much better than the previous "tests" which was essentially a small selenium script. I think we'll stick with this suite for now since its better than nothing, unless of course you'd like to open a PR to contribute more readable tests. :)